<h1 align = "center"> Independent T-test </h1>

* if the samples contain more than about 50 scores the sampling distribution
should be normally distributed.

In [22]:
Group<-gl(2, 12, labels = c("Picture", "Real Spider")) #Generate factors by specifying the pattern of their levels.
Anxiety<-c(30, 35, 45, 40, 50, 35, 55, 25, 30, 45, 40, 50, 40, 35, 50, 55, 65,
55, 50, 35, 30, 50, 60, 39)

In [23]:
spiderLong<-data.frame(Group, Anxiety)

### Test for normality

In [5]:
library(pastecs)

Loading required package: boot


In [6]:
by(spiderLong$Anxiety, spiderLong$Group, stat.desc, basic = FALSE, norm =
TRUE)

spiderLong$Group: Picture
      median         mean      SE.mean CI.mean.0.95          var      std.dev 
  40.0000000   40.0000000    2.6827168    5.9046200   86.3636364    9.2932038 
    coef.var     skewness     skew.2SE     kurtosis     kurt.2SE   normtest.W 
   0.2323301    0.0000000    0.0000000   -1.3939289   -0.5656047    0.9650165 
  normtest.p 
   0.8522870 
------------------------------------------------------------ 
spiderLong$Group: Real Spider
       median          mean       SE.mean  CI.mean.0.95           var 
 50.000000000  47.000000000   3.183765638   7.007420922 121.636363636 
      std.dev      coef.var      skewness      skew.2SE      kurtosis 
 11.028887688   0.234657185  -0.005590699  -0.004386224  -1.459758279 
     kurt.2SE    normtest.W    normtest.p 
 -0.592315868   0.948872904   0.620569431 

### t-test

In [7]:
ind.t.test<-t.test(Anxiety ~ Group, data = spiderLong)
ind.t.test


	Welch Two Sample t-test

data:  Anxiety by Group
t = -1.6813, df = 21.385, p-value = 0.1072
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -15.648641   1.648641
sample estimates:
    mean in group Picture mean in group Real Spider 
                       40                        47 


In [10]:
spiderLong

Group Anxiety
1      Picture      30
2      Picture      35
3      Picture      45
4      Picture      40
5      Picture      50
6      Picture      35
7      Picture      55
8      Picture      25
9      Picture      30
10     Picture      45
11     Picture      40
12     Picture      50
13 Real Spider      40
14 Real Spider      35
15 Real Spider      50
16 Real Spider      55
17 Real Spider      65
18 Real Spider      55
19 Real Spider      50
20 Real Spider      35
21 Real Spider      30
22 Real Spider      50
23 Real Spider      60
24 Real Spider      39

In [11]:
library(tidyr)


Attaching package: 'tidyr'

The following object is masked from 'package:pastecs':

    extract



In [35]:
spiderLong$id = rep(1:(nrow(spiderLong)/2),2)

In [36]:
spiderLong$Group = as.character(spiderLong$Group)

In [60]:
head(spiderLong)

Group Anxiety id
1 Picture      30  1
2 Picture      35  2
3 Picture      45  3
4 Picture      40  4
5 Picture      50  5
6 Picture      35  6

In [61]:
spiderWide = unstack(spiderLong, Anxiety~Group)

In [62]:
spiderWide

Picture Real.Spider
1       30          40
2       35          35
3       45          50
4       40          55
5       50          65
6       35          55
7       55          50
8       25          35
9       30          30
10      45          50
11      40          60
12      50          39

In [45]:
colnames(spiderWide) = c("id", "picture", "real")

In [46]:
ind.t.test<-t.test(spiderWide$real, spiderWide$picture)
ind.t.test


	Welch Two Sample t-test

data:  spiderWide$real and spiderWide$picture
t = 1.6813, df = 21.385, p-value = 0.1072
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.648641 15.648641
sample estimates:
mean of x mean of y 
       47        40 


In [ ]:
* df has digits because it is corrected for heteroskedacistity based on the Welch's t-stest'

In [8]:
library(WRS)

Loading required package: MASS
Loading required package: akima
Loading required package: robustbase

Attaching package: 'robustbase'

The following object is masked from 'package:boot':

    salinity


Attaching package: 'WRS'

The following object is masked from 'package:robustbase':

    hard.rejection

The following object is masked from 'package:MASS':

    ltsreg

The following object is masked from 'package:stats':

    ecdf

The following object is masked from 'package:grDevices':

    bmp



In [40]:
spiderWide

id Picture Real Spider
1   1      30          40
2   2      35          35
3   3      45          50
4   4      40          55
5   5      50          65
6   6      35          55
7   7      55          50
8   8      25          35
9   9      30          30
10 10      45          50
11 11      40          60
12 12      50          39

In [48]:
yuen(spiderWide$real, spiderWide$picture) #default is 20% trimming

$n1
[1] 12

$n2
[1] 12

$est.1
[1] 46.75

$est.2
[1] 40

$ci
[1] -4.429361 17.929361

$p.value
[1] 0.2161433

$dif
[1] 6.75

$se
[1] 5.209309

$teststat
[1] 1.295757

$crit
[1] 2.146035

$df
[1] 13.91372

In [49]:
yuen(spiderWide$real, spiderWide$picture, tr = .1)

$n1
[1] 12

$n2
[1] 12

$est.1
[1] 46.9

$est.2
[1] 40

$ci
[1] -2.35747 16.15747

$p.value
[1] 0.13444

$dif
[1] 6.9

$se
[1] 4.394441

$teststat
[1] 1.570166

$crit
[1] 2.106632

$df
[1] 17.34403

In [51]:
yuenbt(spiderWide$real, spiderWide$picture, nboot = 2000)

[1] "NOTE: p-value computed only when side=T"


$ci
[1] -5.399292 17.869861

$test.stat
[1] 1.193625

$p.value
[1] NA

$est.1
[1] 46.75

$est.2
[1] 40

$est.dif
[1] 6.75

$n1
[1] 12

$n2
[1] 12

In [52]:
pb2gen(spiderWide$real, spiderWide$picture, alpha=.05, nboot=2000, est = mom) #m estimator

[1] "Taking bootstrap samples. Please wait."


$est.1
[1] 47

$est.2
[1] 40

$est.dif
[1] 7

$ci
[1] -3.909091 17.250000

$p.value
[1] 0.2255

$sq.se
[1] 28.9248

$n1
[1] 12

$n2
[1] 12

#### Effect size 

In [53]:
t<-ind.t.test$statistic[[1]]

In [54]:
t

[1] 1.681346

In [55]:
df<-ind.t.test$parameter[[1]]

In [56]:
r <- sqrt(t^2/(t^2+df))

In [57]:
r

[1] 0.3416978

* Report the result

On average, participants experienced greater anxiety from real spiders (M =
47.00, SE = 3.18), than from pictures of spiders (M = 40.00, SE = 2.68).
This difference was not significant t(21.39) = —1.68, p > .05; however, it
did represent a medium-sized effect r = .34

#### Steps for t-test

* The independent t-test compares two means, when those means have come from different groups of
entities; for example, if you have used different participants in each of two experimental conditions.
* Look at the value labelled p-value. If the value is less than .05 then the means of the two groups are
significantly different.
* Look at the values of the means to tell you how the groups differ.
Report the t-statistic, the degrees of freedom and the p-value. Also report the means and their
corresponding standard errors (or draw an error bar chart).
* Calculate and report the effect size.

In [58]:
dep.t.test<-t.test(spiderWide$real, spiderWide$picture, paired = TRUE)
dep.t.test


	Paired t-test

data:  spiderWide$real and spiderWide$picture
t = 2.4725, df = 11, p-value = 0.03098
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
  0.7687815 13.2312185
sample estimates:
mean of the differences 
                      7 


In [59]:
yuend(spiderWide$real, spiderWide$picture)

$ci
[1] -1.843818 15.343818

$p.value
[1] 0.1056308

$est1
[1] 46.75

$est2
[1] 40

$dif
[1] 6.75

$se
[1] 3.634327

$teststat
[1] 1.85729

$n
[1] 12

$df
[1] 7